sklearn/metrics/tests/test_common.py

```diff
@@ -1758,6 +1758,9 @@ def check_array_api_metric(
     with config_context(array_api_dispatch=True):
         metric_xp = metric(y_true_xp, y_pred_xp, sample_weight=sample_weight)
 
+        if not isinstance(metric_xp, float):
+            metric_xp = _convert_to_numpy(metric_xp, xp)
```

Comment:
Doesn't this seem a bit redundant because we are doing _convert_to_numpy below again?

Reply:
good point! I think I added it before the other one was introduced

Reply:
Good point! I think I added this before the other one was introduced